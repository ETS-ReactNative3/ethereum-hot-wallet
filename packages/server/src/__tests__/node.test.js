// @flow
import micro from 'micro';
import listen from 'test-listen';
import request from 'request-promise';
import index from '../index';
import MOCK_NODE from '../__mock-data__/node';
import { type Handler, type Req, type Res } from '../utils/type.flow';

jest.mock('../utils/web3-helper', () => ({
  extend: () => {},
  middleware: () => (handler: Handler) => (
    req: Req,
    res: Res,
    ...restArgs: any
  ) => {
    req.context = {
      ...req.context,
      web3: {
        extend: () => {},
        eth: {
          getTransaction: () => Promise.resolve({}),
          getBlock: () => Promise.resolve({}),
        },
        admin: {
          nodeInfo: () => Promise.resolve(MOCK_NODE),
        },
      },
    };
    return handler(req, res, ...restArgs);
  },
}));

it('should return correct json data', async () => {
  const service = micro(index);
  let error;
  try {
    const url = await listen(service);
    const response = await request({
      uri: `${url}/node`,
    });
    expect(response).toMatchSnapshot();
  } catch (err) {
    error = err;
  }

  expect(error).toBeUndefined();
  service.close();
});
