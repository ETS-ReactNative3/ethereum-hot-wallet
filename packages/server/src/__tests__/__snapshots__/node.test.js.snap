// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return correct json data 1`] = `"{\\"enode\\":\\"enode://a18907c4c4e27e1bb417b098499913b66bd99bd74cd6a4dfd2a315ca23a0f09bb05a6a838f4d0e107c60d19df4c0e41dc34de5a3c30435f5d84904461aa9b@192.168.1.101:30303\\",\\"name\\":\\"Geth/v1.8.13-stable/darwin-amd64/go1.10.3\\"}"`;
